libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -348,10 +364,16 @@ export class JsonSchema {
    */
   public validateObjectWithCallback(
     jsonObject: JsonObject,
-    errorCallback: (errorInfo: IJsonSchemaErrorInfo) => void
+    errorCallback: (errorInfo: IJsonSchemaErrorInfo) => void,
+    options?: IJsonSchemaValidateObjectWithOptions
   ): void {
     this.ensureCompiled();
 
+    if (options?.ignoreSchemaField) {
+      jsonObject = { ...jsonObject };
+      delete jsonObject.$schema;
```

Comment:
```suggestion
      const {
        $schema,
        ...remainder
      } = jsonObject;
      jsonObject = remainder;
```
Let's not force a deopt of the object before we schema validate it?